### PR TITLE
feat: add an way to round object's translation to nearest given value

### PR DESCRIPTION
**Pull request by RebeccaStevens**
This allows for people like me who want things to be aligned perfectly to be able to achieve just that.

Note: This feature is currently only useful for things aligned to NSWE.

This will probably need to be reworked before it is universal enough to be added. I've got plans on how to do this but will require a lot of other work with in the code base first.

~~This can be used to satisfy #147 when setting the `step` size to `800` and applying it only to the x and y axises (This should only be done to foundations).~~ Thinking about it, this doesn't quite do what that issue requests.

### CODE REVIEW

**Comment by AnthorNet** [comment]
Hey, I'll take a look soon, not sure about the UI part?

File: src/Modal/Selection.js [hunk 1]
```diff
@@ -248,6 +249,7 @@ export default class Modal_Selection
 
                 inputOptions.push({group: 'Positioning', text: 'Offset selected items position', value: 'offset'});
                 inputOptions.push({group: 'Positioning', text: 'Rotate selected items position', value: 'rotate'});
+                inputOptions.push({group: 'Positioning', text: 'Round selected items\' positions to nearest N meters', value: 'roundtranslation'});
```
Comment (RebeccaStevens) [inline src/Modal/Selection.js:252]: This should probably be reworded; especially as the units are centimeters not meters.

File: src/Modal/Selection.js [hunk 2]
```diff
@@ -489,6 +491,67 @@ export default class Modal_Selection
         });
     }
 
+    static callbackRoundtranslation(baseLayout, markers)
+    {
+        BaseLayout_Modal.form({
+            title       : 'You have selected ' + markers.length + ' items',
+            message     : 'Step Size is the number to round to. The offset will be applied after the rounding.<br /><strong>NOTE:</strong> A foundation is 800 wide.',
```
Comment (RebeccaStevens) [inline src/Modal/Selection.js:498]: There's probably a better description that could be used here.

File: src/Selection/RoundTranslation.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        this.markers                = options.markers;
+
+        this.step                   = Number.parseFloat(options.step);
+        this.offset                 = Number.parseFloat(options.offset);
+        this.roundX                 = Boolean(options.roundX);
+        this.roundY                 = Boolean(options.roundY);
+        this.roundZ                 = Boolean(options.roundZ);
+
+        this.selectionBoundaries    = (options.selectionBoundaries !== undefined) ? options.selectionBoundaries : null;
+        this.useHistory             = false;
```
Comment (RebeccaStevens) [inline src/Selection/RoundTranslation.js:18]: To enable this to be undo-able, would require a bit of work as all the objects' previous positions would need to be stored.

**Pull request closed without merging**